syntax = "proto3";
package dispatcher;
import "base.proto";


service GatewayReceiver {
  rpc SendOrderStatus(OrderStatus) returns (base.Empty) {};
}

service DispatcherGateway {
  rpc GetCurrentPrice(base.ISIN) returns (base.StockPrice) {};
  rpc GetOpeningPrice(base.ISIN) returns (base.StockPrice) {};
  rpc PlaceOrder(base.Order) returns (base.Empty) {};
  rpc CancelOrder(base.CancelOrderRequest) returns (base.Empty) {};
}

message OrderStatus {
  string token = 1;
  base.Asset asset = 2;
  bool is_buy = 3;
  int64 order_id = 4;
  repeated Transaction transactions = 5;
}

message Transaction {
  double price = 1;
  int32 amount = 2;
}
